src/app/pages/splash/services/splash.service.ts

```diff
@@ -18,7 +19,15 @@ export class SplashService {
   ) {}
 
   evalEnrolment() {
-    return this.token.isValid().catch(() => false)
+    return this.token
+      .refresh()
+      .catch(e => {
+        if (e.status == 401) {
```

Comment:
I think if 401 code is returned, should it not be set to null in all cases? Since 401 means the refresh token is not authorised anymore and cannot refresh the access token?

Reply:
I was thinking to just keep it to the specific use case since the next bit will check the validity of the actual token anyway? Since there might be instances for example when the client secret is accidentally/incorrectly changed, etc?

Reply:
OK sure, I am just wondering if there might be other cases (similar to this one), where the app thinks token is valid and hence is just stuck instead of re-enrolling. But, good point that we don't want to do this for cases where it is not necessary and might fix without a reset/re-enrol. Maybe create an issue so we can discuss the best way to handle these generally to avoid any conditions we might not have explicitly handled.

Reply:
Issue added here #1355